### ISSUE TITLE
MEF composition errors are logged as warnings

### ISSUE THREAD

**Issue by mhutch** [mono/monodevelop#4426]
MEF composition errors are real errors that can cause loss of functionality, but for some reason they are only logged as info, so are not reported to telemetry.

```
INFO [2018-04-03 11:59:35Z]: MEF composition error: Microsoft.CodeAnalysis.Editor.Implementation.Workspaces.EditorTextFactoryService.ctor(textBufferCloneService): expected exactly 1 export of Microsoft.CodeAnalysis.Text.ITextBufferCloneService but found 0.
INFO [2018-04-03 11:59:35Z]: MEF composition error: Microsoft.CodeAnalysis.Editor.Implementation.Peek.PeekableItemSourceProvider.ctor(peekResultFactory): expected exactly 1 export of Microsoft.VisualStudio.Language.Intellisense.IPeekResultFactory but found 0.
```

**Comment by Therzok** [comment]
We're missing EditorFeatures.Text from composition:
https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Ide/ExtensionModel/MonoDevelop.Ide.addin.xml#L409

**Comment by Therzok** [comment]
@KirillOsenkov is there anything else we need to include?

**Comment by Therzok** [comment]
Should we be generating them automatically for things in ReferencesRoslyn/VSEditor.props?

**Comment by KirillOsenkov** [comment]
No, I'd like to have intentional manual control over these. Don't worry, if we were missing something important, composition would tell us ;)

On MS.CA.EF.Text.dll - good catch, but there's only one type exported: `TextBufferCloneServiceFactory`. Apparently it's non-essential. I'll still add it to the composition.
On MS.VS.Language.intelliSense.dll - it doesn't have any MEF exports inside it.

**Comment by Therzok** [comment]
We're still missing IPeekResultFactory. It seems we need to supply:
a) an implementation for IDocumentPeekResult
b) an implementation for IExternallyBrowsablePeekResult
c) an implementation for IPeekResultFactory

**Comment by KirillOsenkov** [comment]
There are Roslyn bugs tracking all of those. I've verified that the failures we're seeing are completely benign (i.e. not causing cascading failures, because these are the leafs in the MEF graph).

We can certainly provide stubs to shut up the warnings, but I'm not sure it's worth it? Let me know and I could do it if you feel it's cleaner this way.
